src/tray/templates/services/app/deploy.yml.j2

```diff
@@ -0,0 +1,90 @@
+apiVersion: v1
+kind: Deployment
+metadata:
+  labels:
+    app: joanie
+    service: joanie
+    version: "{{ joanie_image_tag }}"
+    deployment_stamp: "{{ deployment_stamp }}"
+  name: "joanie-app-{{ deployment_stamp }}"
+  namespace: "{{ namespace_name }}"
+spec:
+  replicas: {{ joanie_app_replicas }}
+  selector:
+    matchLabels:
+      app: joanie
+      service: joanie
+      version: "{{ joanie_image_tag }}"
+      deployment: "joanie-app-{{ deployment_stamp }}"
+      deployment_stamp: "{{ deployment_stamp }}"
+  template:
+    metadata:
+      labels:
+        app: joanie
+        service: joanie
+        version: "{{ joanie_image_tag }}"
+        deployment: "joanie-app-{{ deployment_stamp }}"
+        deployment_stamp: "{{ deployment_stamp }}"
+    spec:
+      # Prefer running pods on different nodes for redundancy
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchExpressions:
+                    - key: deployment
+                      operator: In
+                      values:
+                        - "joanie-app-{{ deployment_stamp }}"
+                topologyKey: kubernetes.io/hostname
+{% set image_pull_secret_name = joanie_image_pull_secret_name | default(none) or default_image_pull_secret_name %}
+{% if image_pull_secret_name is not none %}
+      imagePullSecrets:
+        - name: "{{ image_pull_secret_name }}"
+{% endif %}
+      containers:
+        - name: joanie
+          image: "{{ joanie_image_name }}:{{ joanie_image_tag }}"
+          imagePullPolicy: Always
+          livenessProbe:
+            httpGet:
+              path: /__heartbeat__
+              port: {{ joanie_django_port }}
+              httpHeaders:
+                - name: Host
+                  value: "{{ joanie_host }}"
+            initialDelaySeconds: 60
+            periodSeconds: 30
+          readinessProbe:
+            httpGet:
+              path: /__lbheartbeat__
+              port: {{ joanie_django_port }}
+              httpHeaders:
+                - name: Host
+                  value: "{{ joanie_host }}"
+            initialDelaySeconds: 10
+            periodSeconds: 5
+          env:
+            - name: DB_HOST
+              value: "joanie-{{ joanie_database_host }}-{{ deployment_stamp }}"
+            - name: DB_NAME
+              value: "{{ joanie_database_name }}"
+            - name: DB_PORT
+              value: "{{ joanie_database_port }}"
+            - name: DJANGO_ALLOWED_HOSTS
+              value: "{{ joanie_host }}"
+            - name: DJANGO_CONFIGURATION
+              value: "{{ joanie_django_configuration }}"
+            - name: DJANGO_SETTINGS_MODULE
+              value: joanie.settings
```

Comment:
This change can be postponed, but it's a recurrent issue on all of our trays using "dot env" ConfigMaps.

Reply:
actually no because we override this file in matsuo so it would be useless to set it via Ansible... I think it's easier and a better pattern to separate what is set via Ansible and what is set via the _env.yml.j2 file. The current split is sane in my opinion and we should not change it.

Reply:
I don't get your point as you can set the ` _env.yml.j2` file via Ansible and also override it in matsuo, but let's keep it as is if you think it's a better design.

Reply:
my point was that we always need to override it. So what is set via Ansible doesn't bring any value as we always need to configure it in the _env file... So I find it confusing.

Reply:
:bulb: To prevent overriding it, we can make this template extensible _via_ Jinja or better _via_ Ansible:

```yml
joanie_extra_env_vars:
  - FOO: "bar"
  - SPAM: "lol"
```

But again, this refactoring can be postponed.

Reply:
yep! Or like we do for nginx.